WordPressVIPMinimum/Sniffs/Compatibility/RestrictedCacheGroupSniff.php

```diff
@@ -0,0 +1,99 @@
+	 *
+	 * @var string
+	 */
+	protected $group_name = 'wp_cache_functions';
+
+	/**
+	 * Functions this sniff is looking for.
+	 *
+	 * @var array The only requirement for this array is that the top level
+	 *            array keys are the names of the functions you're looking for.
+	 *            Other than that, the array can have arbitrary content
+	 *            depending on your needs.
+	 */
+	protected $target_functions = [
+		'wp_cache_add' => true,
+		'wp_cache_set' => true,
+	];
+
+	/**
+	 * List of cache group names already in use by wp-memcached.
+	 *
+	 * @var array
+	 */
+	private $wp_memcached_groups = [
+		'blog-details'              => true,
+		'blog-id-cache'             => true,
+		'blog-lookup'               => true,
+		'bookmark'                  => true,
+		'calendar'                  => true,
+		'category'                  => true,
+		'category_relationships'    => true,
+		'comment'                   => true,
+		'counts'                    => true,
+		'general'                   => true,
+		'global-posts'              => true,
+		'options'                   => true,
+		'plugins'                   => true,
+		'post_ancestors'            => true,
+		'post_format_relationships' => true,
+		'post_meta'                 => true,
+		'post_tag_relationships'    => true,
+		'posts'                     => true,
+		'rss'                       => true,
+		'site-lookup'               => true,
+		'site-options'              => true,
+		'site-transient'            => true,
+		'term_meta'                 => true,
+		'terms'                     => true,
+		'themes'                    => true,
+		'timeinfo'                  => true,
+		'transient'                 => true,
+		'user_meta'                 => true,
+		'useremail'                 => true,
+		'userlogins'                => true,
+		'usermeta'                  => true,
+		'users'                     => true,
+		'userslugs'                 => true,
+		'widget'                    => true,
+
+	];
+
+	/**
+	 * Process the parameters of a matched function.
+	 *
+	 * @param int    $stackPtr        The position of the current token in the stack.
+	 * @param array  $group_name      The name of the group which was matched.
+	 * @param string $matched_content The token content (function name) which was matched.
+	 * @param array  $parameters      Array with information about the parameters.
+	 * @return int|void Integer stack pointer to skip forward or void to continue
+	 *                  normal file processing.
+	 */
+	public function process_parameters( $stackPtr, $group_name, $matched_content, $parameters ) {
+		if ( count( $parameters ) > 2 && isset( $this->wp_memcached_groups[ trim( $parameters[3]['raw'], '"\'' ) ] ) ) {
+			$this->phpcsFile->addError( 'Please do not use cache group %s, as it is already in use by wp-memcached: https://docs.wpvip.com/technical-references/caching/object-cache/.', $stackPtr, 'Wp_memcached', $parameters[3]['raw'] );
+		}
```

Comment:
This part needs revisiting.

Let me try and list the issues:
1. Why use `count( $parameters )` when you can use `isset( $parameters[3] )` ? (performance)
2. The `raw` index should rarely be used. If it is used at all, it should generally be combined with a token type check and a non-empty token count check.
    Consider the following code sample which would now also be flagged (but should be):
    ```php
    wp_cache_add( $key, $data, posts ); // OK.
    ```
    (while it may not be common for constants to be lowercase, it is perfectly valid PHP)
    Also consider the following which would now **not** be flagged:
    ```php
    wp_cache_add(
        $cache_key, // Key.
        'data', // Data.
        'posts' // Group name.
    ); // Error.
    ```
    I'd recommend walking the tokens in the parameter instead. Basically, you are looking for text strings and safe for a few exception, if anything else is found, you can bow out without doing the text content comparison.
    Here are some additional tests to consider:
    ```php
    // OK (undetermined).
    wp_cache_add( $key, $data, $name );
    wp_cache_add( $key, $data, 'category' . '_relationships' );
    wp_cache_add( $key, $data, "category_$relationships" );

    // These should be flagged.
    wp_cache_add( $cache_key, 'data', <<<EOD
    posts
    EOD
    );

    wp_cache_add( $cache_key, 'data', <<<'EOD'
    posts
    EOD
    );
    ```
    Again, while those last two are probably very rare, the sniff should be able to handle these.
3. Stripping quotes via `trim()` is risky. I've explained this elsewhere before. Please use the `Sniff::strip_quotes()` function instead.
4. What about using a "bow out early" pattern instead of crammed everything into one condition ? (code readability)
5. Not sure if the error code is very descriptive. What about `MemcachedGroupNameFound` or just plain `Found` ?
6. The `$data` parameter in the `addError()` method expects an array. A single text string is being passed and will cause a fatal error when this sniff is run on PHP 8.0.

Nitpick: for code readability, it would be better for the `addError()` function call to use multi-line format.